### ISSUE TITLE
Questions about the youtube download player

### ISSUE THREAD

**Issue by Karweck** [Rapptz/discord.py#358]
Hey there :)
I am still new to programming , but I hope you guys do not mind helping me out.
I tried coding a music bot with your library and had some trouble with the player that downloads youtube videos to stream them. I got that part working, but whenever a video is playing and another one is being requested (lauching the play_yt() function) the player crashes and I have to restart the bot. My first solution was to check if the audio stream was playing before allowing another link to be opened, but that left me with the following problem: As the player instance is only created when the play_yt() function is called, it doesnt allow me to check if player.is_playing() beforehand, as a player does not exist yet - resulting in an error message and crashing my bot. But I need to check if there is a player running before launching the play_yt() function to avoid crashing the audioplayer. Any thoughts on how I can check if a player is playing, even though there might be no player to begin with? I am using the ffmpeg player. 

The next issue I have is how the yt downloader deals with playlists. When I enter a playlist as a link in create_ytdl_player(link) it starts downloading the whole playlist, but only ever plays the first video in it. Is there a way to stream the entire playlist? Also it crashes completly whenever a video in a playlist is not available in my country, is there any way to prevent that?

I hope my explanation was not too confusing and you might be able to help me. I added the important code snippets for a better understanding.

` 

```
async def play_yt(link):

    global player  # ugly,  but I need to access the player from many other functions
    player = await client.voice_client_in(client.get_server(serverID)).create_ytdl_player(link)
    player.volume = 0.05
    player.start()`
```

`      

```
   elif message.content.startswith('!yt'):
       link = message.content.strip('!yt ')
       await play_yt(link)`
```

**Comment by ccubed** [comment]
1. If you're ever using python and find yourself saying "_You know what I need? I need to use global._" Stop immediately, pick up your keyboard and hit yourself with it. You are travelling down a path that is wrong.

2. You're using a global player, so every time play_yt gets called, your bot crashes because it plays hackey sack with your memory regions and process management.

3. You're going to need to check if a player exists somehow, or attach it to individual servers with voice clients. Maybe you'd use Try...Except or maybe you'd just assume that if voice_client_in doesn't return a voice client, they can't be listening to music and don't have a player.

4. YTDL downloads the information for playlists, but the code doesn't push all those links to the ffmpeg player, you'll have to do that yourself.

5. Prevent crashing? Try...Except sounds like it might be helpful.

**Comment by tkdberger** [comment]
ELi5 number one please. Just curious.

**Comment by ButterCheezii** [comment]
Try this?

```py
async def play_yt(link):
    player = await client.voice_client_in(client.get_server(serverID)).create_ytdl_player(link)
    player.volume = 0.05
    player.start()
    return player
```

```py
    elif message.content.startswith('!yt'):
        link = message.content.strip('!yt ')
        player = await play_yt(link)
        # do whatever you need to do with the player here
```
